src/main/java/org/jenkinsci/maven/plugins/hpi/HpiMojo.java

```diff
@@ -136,11 +140,23 @@ private void performPackaging()
             jarArchiver.addConfiguredManifest(manifest);
             File indexJelly = new File(getClassesDirectory(), "index.jelly");
             if (!indexJelly.isFile()) {
-                throw new MojoFailureException("Missing " + indexJelly + ". Delete any <description> from pom.xml and create src/main/resources/index.jelly:\n" +
-                    "<?jelly escape-by-default='true'?>\n" +
-                    "<div>\n" +
-                    "    The description here…\n" +
-                    "</div>");
+                if (getProjectDescription() != null) {
+                    getLog().warn("src/main/resources/index.jelly does not exist. A default one will be created using the description of the pom.xml");
+                    try (final FileOutputStream fos = new FileOutputStream(indexJelly);
+                         final OutputStreamWriter indexJellyWriter = new OutputStreamWriter(fos, StandardCharsets.UTF_8)) {
+                        indexJellyWriter.write("<?jelly escape-by-default='true'?>\n" +
+                                "<div>\n" +
+                                project.getDescription() + "\n" +
```

Comment:
You would need to escape any XML metacharacters here.

Reply:
Some metacharacters like `&`, `<` and `>` can't be put in the POM's `<description>` as the build will failed.

For other ones, like `"` and `'`, the `StringEscapeUtils#escapeXml(String)` doesn't replace them with their equivalent. So not sure if you are thinking about something else.

Reply:
> can't be put in the POM's `<description>`

Escaped they can, and `Project.getDescription` will return the raw text.

Reply:
Got it. Updated